query-planner-js/src/buildPlan.ts

```diff
@@ -4376,44 +4383,23 @@ function computeGroupsForTree(
             );
             updated.group = requireResult.group;
             updated.path = requireResult.path;
-
-            // add __typename to site where we are retrieving context from
-            // this is necessary because the context rewrites path will start with a type condition
-            if (contextToSelection) {
-              assert(isCompositeType(edge.head.type), () => `Expected a composite type for ${edge.head.type}`);
-              updated.group.addAtPath(path.inGroup().concat(new Field(edge.head.type.typenameField()!)));
-            }
-
+            
             if (contextToSelection) {
-              const newContextToConditionsGroups = new Map<string, FetchGroup[]>([...contextToConditionsGroups]);
-              for (const context of contextToSelection) {
-                newContextToConditionsGroups.set(context, [group]);
+              if (requireResult.createdGroups.length === 0) {
```

Comment:
Was curious about why we had this `if` block, synced with @clenfest but we didn't see a reason here.